### ISSUE TITLE
For syncing many hundreds of thousahds of files, Mega fails

### ISSUE THREAD

**Issue by tio-trom** [meganz/MEGAsync#671]
I wanted to share my experience with Magasync Client on Linux (Arch). Tried for months now to sync some 8TB of data, and millions of files. The client freezes so often I have to kill the process and restart it again. But after a while it simple cannot sync anymore I get a 
![2022-02-14_21-07](https://user-images.githubusercontent.com/38043315/153938405-d15bb471-a416-4fdf-89df-79e97dfc6923.png)

I reduced the 8TB of files to some 4TB and from millions to hundreds of thousands. Same thing...

I am tired of trying. I contacted the support so many times and they could never fix it. I would really like to use Mega, but I can't.

If there is a solution to this let me know.

**Comment by angelds-mega** [comment]
Hello, can you please email support@mega.nz so that we can investigate further. Thank you.

**Comment by tio-trom** [comment]
Yes I have done that many times but nothing has fixed the issue...so I thought to open a public issue about it. Maybe others can help out with other ideas/debugging.

**Comment by angelds-mega** [comment]
Do you mind posting the ticket id number here so we can check it for you?

**Comment by angelds-mega** [comment]
Yes, that's correct. We found the ticket made a follow up with our developers. We will send you an email as soon as we heard back from our developers. Thank you for your patience.

**Comment by tio-trom** [comment]
Thanks. I've heard that before tho and unfortunately I wasn't contacted back and nothing was fixed....I wish I could use Mega to backup all of my files properly...

**Comment by GH-A-CC** [comment]
@tio-trom  For what its worth I have found it can take _**hours**_ sitting on  "Downloading File List" and "Scanning" statuses before synchronization starts. Normally the latter - and this is for the largest of folders with less than a million files and less than 1TB in size. If I do a "Force a Full Scan" and restart MegaSync, I think it typically takes something like >2 hours before any actual sync starts. 

However, the ~1TB folder is on a drive connected over USB-3.1 and assuming the drive has a consistent read-spead of ~130MB/sec, I would guess that MegaSync is reading all the data and doing a check-sum/hash validation before anything else,  and thus nearly all that time is simply local file access delays.

1TB = ~1048576 MB  divided by 135 MB/sec transfer speed = ~7767 seconds = ~129 mins = 2 hours 9 mins.

The fact it doesn't do any sync work before checking *everything* seems to be the problem. There should be an option for  MegaSync to start synchronising as soon as it can for a rather than checking everything first

If you have a sync for  /folder/ to mega:/Cloud_Drive/Folder/ where /folder/ has three sub folders (e.g. /folder/A/, /folder/B/, /folder/C/),  there should be an option for  MegaSync to start synchronising /folder/A as soon as it can - e.g. updating /folder/A (while also checking /folder/B and /folder/C) rather than having to check all of /folder/ before any sync work is started.

**Comment by mattw-mega** [comment]
Hi @tio-trom , you will hopefully be pleased to know we have a big update coming up which may help with this.  Since the sync logic and scanning/fingerprinting run on separate threads, you may find MEGAsync much more responsive and it may start uploading earlier. You can read about it and perhaps help test it here: https://github.com/meganz/MEGAsync/issues/770 thanks

**Comment by DanielJoseDolz** [comment]
> I wanted to share my experience with Magasync Client on Linux (Arch). Tried for months now to sync some 8TB of data, and millions of files. The client freezes so often I have to kill the process and restart it again. But after a while it simple cannot sync anymore I get a ![2022-02-14_21-07](https://user-images.githubusercontent.com/38043315/153938405-d15bb471-a416-4fdf-89df-79e97dfc6923.png)
> 
> I reduced the 8TB of files to some 4TB and from millions to hundreds of thousands. Same thing...
> 
> I am tired of trying. I contacted the support so many times and they could never fix it. I would really like to use Mega, but I can't.
> 
> If there is a solution to this let me know.

Same here

**Comment by DanielJoseDolz** [comment]
> Hi @tio-trom , you will hopefully be pleased to know we have a big update coming up which may help with this. Since the sync logic and scanning/fingerprinting run on separate threads, you may find MEGAsync much more responsive and it may start uploading earlier. You can read about it and perhaps help test it here: #770 thanks

rsync in linux and robocopy in windows have this issue solved. I guess you can solve it too. So far it seems a waste of money and time until the fix

**Comment by tio-trom** [comment]
You guys can try their backup option. It solves those issues. It is a one way sync. Works for me now.

**Comment by DanielJoseDolz** [comment]
> You guys can try their backup option. It solves those issues. It is a one way sync. Works for me now.

I did not know this option exists. Nice. Will try. When the desktop client responds. Apparently it is not great when the number of files exceds 400.000

**Comment by mattw-mega** [comment]
Hi All on this thread.   We have a major update coming up, MEGAsync 5.0 which is much better with huge numbers of files and folders.   You can try it out at #770 and give us feedback on whether it's better for your case or not,  if you like.   thanks

**Comment by DanielJoseDolz** [comment]
So when is the update coming out?

**Comment by pearcemichal** [comment]
Just wanted to chime in and say this is still an issue for me as well. After syncing for awhile, the app will freeze up, and from then on will freeze seconds after starting every time. if I can open the app and pause syncing in the short window before it freezes, I can change settings, but if not the entire app becomes unresponsive. 

The total number of files I am trying to sync is 9015. 

I also tried syncing to the new S4 object storage instead, same issue.

**Comment by pearcemichal** [comment]
Actually, I think the issue I am having is different from the original. In my case, it will start syncing files immediately, but then freeze after awhile.

However, I think I may have found a clue. I noticed that the sync would always freeze right when it got to the "finalizing" stage on a specific file: 03 Allegro Vivace.mp3. I am syncing my music library, and in this case specifically with classical music, there are several files called 03 Allegro Vivace.mp3 under different sub folders. (It's a common naming convention in classical music). 

All of my classical music is in a subfolder of my music library, so as a test I tried just adding that folder to the megaignore and rebooted the sync. It was then able to finish syncing without issues. 

As a more definitive test, I am going to try mass renaming all of my classical music files to something like <composer>-<composition>-<name>, to guarantee the file names are unique, and try syncing it again.

**Comment by rl-mega** [comment]
Hi @pearcemichal, your issue might be of a different order, you are right. There should be no issue with non-unique filenames though. Your description make me think that you might have an issue with the media processing library (we use libmediainfo to extract information such as track length).
There is a command line option to disable this processing. Could you
- exit megasync
- open a powershell terminal
- start megasync with `<path to megasync>\megasync.exe /nogfx`? If you installed the app for your user only (default install), that would be: `& $env:LOCALAPPDATA\MEGAsync\MEGAsync.exe /nogfx`
- remove the folder from megaignore
- check that the folder gets synchronized without issue
- next time you can start megasync without the option
I will close this ticket because it is not related, but you can still report here if it fixed your issue. If it did not, please open a new ticket.

Thanks in advance and have a good day.
